docs/theme.md

```diff
@@ -0,0 +1,138 @@
+
+```jsx
+// IconButton.js
+
+import React from 'react'
+import { ThemeConsumer } from 'contexture-react/src/utils/theme'
+
+let IconButton = ({ icon, children, ...props }) => (
+  <ThemeConsumer>
+    {theme =>
+      <theme.Button {...props}>
+        <theme.Icon icon={icon} />
+        {children}
+      </theme.Button>
+    }
+  </ThemeConsumer>
+)
+export default IconButton
+```
+
+### withTheme
+
+The `withTheme` higher-order component injects the theme object from context into its component argument as the `theme` prop. Here is the same IconButton example using `withTheme`:
+
+```jsx
+// IconButton.js
+
+import React from 'react'
+import { withTheme } from 'contexture-react/src/utils/theme'
+
+let IconButton = ({ theme, icon, children, ...props }) => (
+  <theme.Button {...props}>
+    <theme.Icon icon={icon} />
+    {children}
+  </theme.Button>
+)
+export default withTheme(IconButton)
+```
+
+`ThemeConsumer` and `withTheme` both accomplish the same thing in different ways. You're free to use whichever one you like best. 🙂
+
+
+## Default theme components
+
+The default theme in contexture-react is GreyVest. The full list of **theme components** included in GreyVest is shown in the table below.
+
+| Key | Expected props | Notes |
+| --- | --- | --- |
+| `AdderPicker` | `options`, `onChange`, `label` | *this key is actually based on specific usage in the FilterAdder component, and should probably just be removed from the defaults in favor of a nested style* |
+| `BarChart` | `height`, `borderColor`, `min`, `max` | Used in the DateHistogram and TermsStats example types |
+| `Box` | `children` | A generic container element |
+| `Button` | `isActive`, `primary`, `onClick`, `children` | A generic button |
+| `Checkbox` | `checked`, `onChange` | A generic checkbox |
+| `CheckButton` | `checked`, `onClick`, `children` | A button with an `onClick` handler and an on/off state, containing a checkbox that displays this state *(this really should be extrapolated to some kind of generic BinaryStateButton)* |
+| `ExpandableTable` | `data`, `columns`, `recordKey`, `sortField`, `sortDir`, `columnSort` | Used in the TermsStatsTable and CheckableTermsStatsTable example types |
+| `Icon` | `icon`, `onClick` | A generic icon component |
+| `Fonts` | none | For holding `<link>` elements to fonts used in the theme |
+| `Input` | same as basic `input` | A generic input component |
+| `ListItem` | `children` | A generic list item |
+| `MissingTypeComponent` | `node` | Used as a fallback component in search interfaces when no other component is found for a node's contexture type |
+| `Modal` | `isOpen`, `children` | A generic modal component |
+| `PagerItem` | `active`, `disabled` | A list item for Pager components *(make this nested instead?)* |
+| `Picker` | `options`, `onChange` | Renders a list of selectable options |
+| `PickerItem` | `active`, `disabled` | A list item for Picker components *(make this nested instead?)* |
```

Comment:
It occurs to me that if PickerItem is only every used by Picker (as I imagine it is), perhaps it should just be an intrinsic part of `Picker`. Does it really make sense to override PickerItem but not Picker, or Picker but not PickerItem?

Reply:
> Does it really make sense to override PickerItem but not Picker, or Picker but not PickerItem?

I think it actually does - in fact, the blueberry theme overrides only PickerItem (to be the same as its regular ListItem component), and doesn't touch Picker itself. Overriding the picker component lets you add functionality (eg modals and nesting), while the item component is more of a styling thing, so they're different concerns.

(As a side note, this is basically how it was with TextHighlight and HighlightWrap; I nixed the `HighlightWrap` key anyway because they're so tiny that it's trivial to declare both at once.)

With `Picker.ListItem` rather than `PickerItem`, you'd only have to override the `ListItem` key in a case like blueberry's, rather than having to remember to do `PickerItem` (and whatever `OtherItem`) too. It's nice, but is it worth making sure that every picker is named `Picker` for withTheme?